frontend/packages/kubevirt-plugin/src/k8s/wrapper/vm/vm-wrapper.ts

```diff
@@ -176,35 +175,19 @@ export class VMWrapper extends K8sResourceWrapper<VMKind, VMWrapper> implements
   ensureDataVolumeTemplates = () => this.ensurePath('spec.dataVolumeTemplates', []);
 
   private ensureStorages = () => {
-    if (_.isEmpty(this.getDisks())) {
-      this.data.spec.template.spec.domain.devices.disks = [];
-    }
-    if (_.isEmpty(this.getVolumes())) {
-      this.data.spec.template.spec.volumes = [];
-    }
-    if (_.isEmpty(this.getDataVolumeTemplates())) {
-      this.data.spec.dataVolumeTemplates = [];
-    }
+    this.ensurePath('spec.template.spec.domain.devices.disks', []);
+    this.ensurePath('spec.template.spec.volumes', []);
+    this.ensurePath('spec.dataVolumeTemplates', []);
   };
 
   private ensureNetworksConsistency = () => {
-    if (_.isEmpty(this.getInterfaces())) {
-      delete this.data.spec.template.spec.domain.devices.interfaces;
-    }
-    if (_.isEmpty(this.getNetworks())) {
-      delete this.data.spec.template.spec.networks;
-    }
+    this.clearIfEmpty('spec.template.spec.domain.devices.interfaces');
+    this.clearIfEmpty('spec.template.spec.networks');
   };
 
   private ensureStorageConsistency = () => {
-    if (_.isEmpty(this.getDisks())) {
-      delete this.data.spec.template.spec.domain.devices.disks;
-    }
-    if (_.isEmpty(this.getVolumes())) {
-      delete this.data.spec.template.spec.volumes;
-    }
-    if (_.isEmpty(this.getDataVolumeTemplates())) {
-      delete this.data.spec.dataVolumeTemplates;
-    }
+    this.clearIfEmpty('spec.template.spec.domain.devices.disks');
```

Comment:
can't we merge `ensureStorages` and `ensureStorageConsistency` into one method?

Reply:
no, one is called before the change of data occurs and the other is called after the change of data - please see `prependStorage` method